crates/apub/src/objects/post.rs

```diff
@@ -226,9 +227,12 @@ impl Object for ApubPost {
 
     let url_blocklist = get_url_blocklist(context).await?;
 
-    if let Some(url) = &url {
+    if let Some(ref mut url) = url {
       is_url_blocked(url, &url_blocklist)?;
       is_valid_url(url)?;
+      if let Some(local_url) = to_local_url(url.as_str(), context).await {
+        *url = local_url;
+      }
```

Comment:
Instead of changing a ref mut, you can do `let url = if let Some....`. It also looks like `to_local_url` returns an option anyway.